File: h264/src/main/java/org/jcodec/codecs/h264/H264Utils2.java
package org.jcodec.codecs.h264;

public class H264Utils2 {

    public static int golomb2Signed(int val) {
        int sign = ((val & 0x1) << 1) - 1;
        val = ((val >> 1) + (val & 0x1)) * sign;
        return val;
    }

}

File: src/main/java/org/jcodec/codecs/h264/H264Utils2.kt
package org.jcodec.codecs.h264

object H264Utils2 {
    fun golomb2Signed(v: Int): Int {
        var _v = v
        val sign = (_v and 0x1 shl 1) - 1
        _v = ((_v shr 1) + (_v and 0x1)) * sign
        return _v
    }
}